pkg/cloudprovider/aws/instance.go

```diff
@@ -105,6 +128,37 @@ func (p *InstanceProvider) Terminate(ctx context.Context, node *v1.Node) error {
 	return nil
 }
 
+// The intention is to remove Offerings from the provided possible InstanceTypes based on recently observed ICEs, which will
+// indirectly impact the packer to keep it from spinning its wheels on packings that are doomed to fail due to EC2 capacity constraints.
+func (p *InstanceProvider) DiscardICEdInstanceTypes(ctx context.Context, instanceTypes []InstanceType) []cloudprovider.InstanceType {
+	var cleanedInstanceTypes = []cloudprovider.InstanceType{}
+	for index, instanceType := range instanceTypes {
+		odIceZones, hasODIces := p.iceCache.Get(createIceCacheKey(instanceType.Name(), v1alpha1.CapacityTypeOnDemand))
+		spotIceZones, hasSpotIces := p.iceCache.Get(createIceCacheKey(instanceType.Name(), v1alpha1.CapacityTypeSpot))
+		if !hasODIces && !hasSpotIces {
+			cleanedInstanceTypes = append(cleanedInstanceTypes, &instanceTypes[index])
```

Comment:
why are you grabbing the instanceType from the slice using an index here but below you are using the iterator variable below for offering?  I assumed it was to get around the iterator loop variable from having the value overridden but it would still point to the same memory address?  Does that not apply below in the offering loop?